include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -600,6 +600,79 @@ using __repacked_tuple_t = typename __repacked_tuple<T>::type;
 template <typename _ContainerOrIterable>
 using __value_t = typename __internal::__memobj_traits<_ContainerOrIterable>::value_type;
 
+//-----------------------------------------------------------------------
+// future and helper classes for async pattern/algorithm
+//-----------------------------------------------------------------------
+
+// empty base class for type erasure
+struct __tmp_base
```

Comment:
Currently we have memory leak. You store the base class in the `std::unique_ptr`. But you don't have virtual destructor. So, for all `Ts...` in `TempObjs` destructors are not called